quantum/rgblight.c

```diff
@@ -1112,3 +1124,20 @@ void rgblight_effect_alternating(animation_status_t *anim) {
   anim->pos = (anim->pos + 1) % 2;
 }
 #endif
+
+#if defined(RGBLIGHT_EFFECT_WPM_TEMP) && defined(VELOCIKEY_ENABLE)
+void rgblight_effect_wpm_temp(animation_status_t *anim) {
+  static uint8_t typing_speed = 0;
+
+  // (slightly) smoother transitions
+  if (typing_speed > get_typing_speed()) {
+    typing_speed--;
+  } else {
+    typing_speed=get_typing_speed();
+  }
+
+  LED_TYPE tmp_led;
+  sethsv(201 - MAX(typing_speed, 80), rgblight_config.sat, rgblight_config.val, &tmp_led);
```

Comment:
It would be awesome if we could add the smoothing/capping functionality into the Velocikey API rather than exposing the typing speed value directly like here, since that would keep all the typing-speed related code neatly in one place (and make this effect code a little neater).

If you were interested in trying to add these capabilities into Velocikey itself, I'm happy to help out, but if you'd rather just merge this code as-is, I can always give it a go myself at a later date :)

What's most important of course is that the transitions created by the code _feel_ right to you, because that's what I found to be the hardest part of writing Velocikey in the first place!

Reply:
Looks like the capping is already there, I can work in the smoothing though! For velocikey proper I'd want to do it a little nicer, though there's a ton of ways to go about that. I'm leaning towards making `typing_speed` a floating point value and applying some kind of decay proportional to its value but I'm open to suggestions

Reply:
No floats pls. This has to run on AVR and be performant. Look into using the lib8tion math library to handle scaling and capping values.